idea/tests/org/jetbrains/kotlin/idea/codeInsight/smartEnter/SmartEnterTest.kt

```diff
@@ -1405,6 +1405,81 @@ class SmartEnterTest : KotlinLightCodeInsightFixtureTestCase() {
 <caret>"""
     )
 
+    fun testValueArgumentList1() = doFileTest(
+        """
+        fun foo(i: Int) = 1
+        fun test1() {
+            foo(1<caret>
+        }
+        """,
+        """
+        fun foo(i: Int) = 1
+        fun test1() {
+            foo(1)<caret>
+        }
+        """
+    )
+
+    fun testValueArgumentList2() = doFileTest(
+        """
+        fun foo(i: Int) = 1
+        fun test2() {
+            foo(foo(1<caret>
+        }
+        """,
+        """
+        fun foo(i: Int) = 1
+        fun test2() {
+            foo(foo(1))<caret>
+        }
+        """
+    )
+
+    fun testValueArgumentList3() = doFileTest(
+        """
+        fun foo(i: Int) = 1
+        fun test3() {
+            foo(<caret>
+        }
+        """,
+        """
+        fun foo(i: Int) = 1
+        fun test3() {
+            foo(<caret>)
+        }
+        """
+    )
+
+    fun testValueArgumentList4() = doFileTest(
+        """
+        fun foo(i: Int) = 1
+        fun test4() {
+            foo(1,<caret>
+        }
+        """,
+        """
+        fun foo(i: Int) = 1
+        fun test4() {
+            foo(1,<caret> )
```

Comment:
This test looks strange. Please try this example as well, the caret is shifted after smart enter, and it doesn't look as expected behavior. 

```
fun foo(i: Int) = 1
fun test4() {
    foo(1, <caret>
}
```